src/Microsoft.PowerShell.Commands.Utility/commands/utility/New-Object.cs

```diff
@@ -199,6 +197,27 @@ protected override void BeginProcessing()
                     }
                 }
 
+                if (SystemPolicy.GetSystemLockdownPolicy() == SystemEnforcementMode.Enforce)
+                {
+                    switch (Context.LanguageMode)
+                    {
+                        case PSLanguageMode.NoLanguage:
+                        case PSLanguageMode.RestrictedLanguage:
+                        if (!CoreTypes.Contains(type))
+                        {
+                            ThrowTerminatingError(
+                                new ErrorRecord(
+                                    new PSNotSupportedException(
+                                        string.Format(NewObjectStrings.CannotCreateTypeLanguageMode, Context.LanguageMode.ToString())),
+                                    nameof(NewObjectStrings.CannotCreateTypeLanguageMode),
+                                    ErrorCategory.PermissionDenied,
+                                    targetObject: null));
+                        }
+
+                        break;
+                    }
+                }
```

Comment:
Sorry I realised I had a slight tweak to this, but might have gotten in too late before:

```suggestion
                switch (Context.LanguageMode)
                {
                    case PSLanguageMode.NoLanguage:
                    case PSLanguageMode.RestrictedLanguage:
                        if (SystemLockdownPolicy.GetSystemLockdownPolicy() == SystemEnforcementMode.Enforce
                            && !CoreTypes.Contains(type))
                        {
                            ThrowTerminatingError(
                                new ErrorRecord(
                                    new PSNotSupportedException(
                                        string.Format(NewObjectStrings.CannotCreateTypeLanguageMode, Context.LanguageMode.ToString())),
                                    nameof(NewObjectStrings.CannotCreateTypeLanguageMode),
                                    ErrorCategory.PermissionDenied,
                                    targetObject: null));
                        }
                        break;
                    }
```